catalog-rest-service/src/main/resources/ui/src/components/MyDataHeader/MyDataHeader.component.tsx

```diff
@@ -93,6 +93,12 @@ const MyDataHeader: FunctionComponent<Props> = ({
         link: `/teams`,
         dataTestId: 'terms',
       },
+      thesaureses: {
+        icon: Icons.TERMS,
+        data: `${entityCounts.thesaurusCount} Thesauruses`,
```

Comment:
Hi @pmbrull, I am slowly getting familiar with the GUI codebase, but I am currently stuck. For some reason the `entityCounts.thesaurusCount` remains `0`, but data is present and the `/api` works. I cannot find the location of the Javascript hook which reads the API to obtain the number of Thesaurus entities. Can you give me a hint?

Reply:
@darth-coder00 @Sachin-chaurasiya Can you guys please give some pointers here to @tomtor?

Reply:
@shahsank3t sure.

Reply:
@tomtor to get thesaurusCount first you will have to add `thesaurusCount` to this [EntityCounts type ](https://github.com/open-metadata/OpenMetadata/blob/f5ec367335992c8dad818c7a29ed77c4fa7bf7fc/catalog-rest-service/src/main/resources/ui/src/interface/types.d.ts#L333)

After that, you have to add `thesaurusCount` to this [ EntityUtil ](https://github.com/open-metadata/OpenMetadata/blob/f5ec367335992c8dad818c7a29ed77c4fa7bf7fc/catalog-rest-service/src/main/resources/ui/src/utils/EntityUtils.tsx#L215) method and based on entity type you can add the counts.

Reply:
@Sachin-chaurasiya Thanks for the feedback. I did both (days ago), but still `0`. For some reason the `Buckets` have no matching entity. Where are the `Buckets` filled?

Reply:
@tomtor You can refer this from here we are calling [getEntityCountByType](https://github.com/open-metadata/OpenMetadata/blob/f5ec367335992c8dad818c7a29ed77c4fa7bf7fc/catalog-rest-service/src/main/resources/ui/src/pages/MyDataPage/MyDataPage.component.tsx#L59) method.

Reply:
@tomtor All those entity counts that we are getting is from search API.

Reply:
@Sachin-chaurasiya Ah thanks, I have to check the `/search` API implementation. I missed that, will have a look at that.